doc/source/general/cpu_hardware.rst

```diff
@@ -142,40 +146,134 @@ Triggers are all of the following:
 -  ON condition { some commands }.
 -  WHEN condition THEN { some commands }.
 
-.. note::
-
-    The :ref:`WAIT <wait>` command only causes mainline code
-    to be suspended.  Trigger code such as WHEN, ON, LOCK STEERING,
-    and LOCK THROTTLE, will continue executing while your program
-    is sitting still on the WAIT command.
+The way these work is that once per **physics tick**, all these
+trigger routines get run, including those locks that are always
+re-evaluated by the cooked steering, and the ``ON`` and ``WHEN``
+triggers.  (This isn't *quite* true.  The real answer is more
+complex than that - see :ref:`CPU Update Loop <cpu_update_loop>`
+elsewhere on this page).
+
+Each of the steering locks behaves like a function that returns
+a value, and is re-called to get the new value for this **physics
+tick**.  Each of the ``ON`` and ``WHEN`` triggers also behave
+much like a function, with a body like this::
+
+   if (not conditional_expression)
+       return 1.  // premature quit.  preserve and try again next time.
+   do_rest_of_trigger_body_here.
+
+
+.. _trigger_conditional:
+
+Triggers always execute at least as far as the Conditional Check
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Even a trigger who's condition isn't true yet still needs to execute
+a few instructions into the trigger subroutine to *discover* that its
+condition isn't true yet.  The trigger still causes a subroutine call
+once per **physics tick** just to get far enough into the routine to
+reach the conditional expression check and discover that it's not
+time to run the rest of the body yet, so it returns.  An expensive
+to calculate conditional expression can really starve the system of
+instructions becuase it's getting run every single **physics tick**.
+*It's good practice to try to keep your trigger's conditional check
+short and fast to execute.  If it consists of multiple clauses, try
+to take advantage of :ref:`short circuit boolean <short_circuit>`
+logic by putting the fastest part of the check first.*
+
+.. _wait_in_trigger:
+
+Wait in a Trigger
+~~~~~~~~~~~~~~~~~
+
+It is possible for kOS to allow a trigger that takes longer than one
+*physics tick* to execute.  It just means the rest of the program is
+stuck until the trigger is done.  Triggers can interrupt mainline code, but
+mainline code can't interrupt triggers.  Thus using a ``wait`` in a trigger,
```

Comment:
This is the only instance where `wait` is lower case.

As a secondary note, it looks like all of the explanations use `wait dt.` as the example, but don't specifically show that `wait until condition.` will do the same thing.  While I think it *should* be clear from the generic use of the term `WAIT`, I can see that being a user question.  I expected this behavior based on the docs, but tested to be sure.